src/cargo/sources/path.rs

```diff
@@ -329,6 +331,29 @@ impl<'cfg> PathSource<'cfg> {
         }
     }
 
+    fn list_files_walk_except_dot_files_and_dirs(
+        &self,
+        pkg: &Package,
+        filter: &mut dyn FnMut(&Path) -> CargoResult<bool>,
+    ) -> CargoResult<Vec<PathBuf>> {
+        let root = pkg.root();
+        let mut exclude_dot_files_dir_builder = GitignoreBuilder::new(root);
+        exclude_dot_files_dir_builder.add_line(None, ".*")?;
+        exclude_dot_files_dir_builder.add_line(None, "*/.*/*")?;
```

Comment:
Is this line necessary?

Reply:
good catch, it is already covered with the other pattern